vimeo-networking/src/main/java/com/vimeo/networking/model/BaseResponseList.java

```diff
@@ -37,6 +49,92 @@
     public Paging paging;
     public ArrayList<T> data;
 
-    // TODO: maybe don't need
     public abstract Class getModelClass();
+
+    public static abstract class BaseResponseTypeAdapter<S extends BaseResponseList> extends TypeAdapter<S> {
+
+        @NotNull
+        private final PagingTypeAdapter mPagingTypeAdapter;
+        @NotNull
+        private final TypeAdapter mTypeAdapter;
+
+        public BaseResponseTypeAdapter(@NotNull PagingTypeAdapter pagingTypeAdapter,
+                                       @NotNull TypeAdapter typeAdapter) {
+            mPagingTypeAdapter = pagingTypeAdapter;
+            mTypeAdapter = typeAdapter;
+        }
+
+        protected abstract S createResponseListObject();
+
+        @Override
+        public void write(JsonWriter out, S value) throws IOException {
+            out.beginObject();
+            if (value == null) {
+                ClientLogger.d("BaseResponseList null in write()");
+                out.endObject();
+                return;
+            }
+
+            out.name("total").value(value.total);
+            out.name("page").value(value.page);
+            out.name("per_page").value(value.perPage);
```

Comment:
Without writing some kind of craziness that writes these TypeAdapters for us, then I am not sure we would be able to get around the monotony and tediousness of writing these methods by hand. I know it sucks. Every article I have come across confirms that it sucks. But if you want type adapters, then we have to go this route.

I could go through and rip out the strings into constants, if you think it warrants that.